packages/backend-common/src/reading/BitbucketUrlReader.ts

```diff
@@ -14,71 +14,249 @@
  * limitations under the License.
  */
 
-import fetch, { RequestInit, HeadersInit, Response } from 'node-fetch';
 import { Config } from '@backstage/config';
-import { ReaderFactory, UrlReader } from './types';
+import parseGitUri from 'git-url-parse';
+import fetch, { HeadersInit, RequestInit, Response } from 'node-fetch';
 import { NotFoundError } from '../errors';
+import { ReaderFactory, UrlReader } from './types';
 
-type Options = {
-  // TODO: added here for future support, but we only allow bitbucket.org for now
+/**
+ * The configuration parameters for a single Bitbucket API provider.
+ */
+export type ProviderConfig = {
+  /**
+   * The host of the target that this matches on, e.g. "bitbucket.com"
+   */
   host: string;
-  auth?: {
-    username: string;
-    appPassword: string;
-  };
+
+  /**
+   * The base URL of the API of this provider, e.g. "https://api.bitbucket.org/2.0",
+   * with no trailing slash.
+   *
+   * May be omitted specifically for Bitbucket Cloud; then it will be deduced.
+   *
+   * The API will always be preferred if both its base URL and a token are
+   * present.
+   */
+  apiBaseUrl?: string;
+
+  /**
+   * The base URL of the raw fetch endpoint of this provider, e.g.
+   * "https://api.bitbucket.org/2.0", with no trailing slash.
+   *
+   * May be omitted specifically for Bitbucket Cloud; then it will be deduced.
+   *
+   * The API will always be preferred if both its base URL and a token are
+   * present.
+   */
+  rawBaseUrl?: string;
+
+  /**
+   * The authorization token to use for requests to this provider.
+   *
+   * If no token is specified, anonymous access is used.
+   */
+  token?: string;
+
+  /**
+   * Username
```

Comment:
Is username/password an alternative auth method to tokens? Should probably mention here in the comments how they are used

Reply:
Yes, you can either authenticate with a personal access token, your regular username,password or your username and personal access token. 
I will clarify this in the documentation and also add some extra tests around the different configuration options

Reply:
We there are other good options (which there seems to be), we should not support username/password based auth. Is there any particular reason you need that?

Reply:
No, only for backward compatibility with the current solution. I will just verify that Bitbucket cloud supports personal access tokens. If that is the case I would be happy to remove those options

Reply:
I have added some comments for what the different properties are used for.